ProcessLib/Deformation/MaterialForces.h

```diff
@@ -0,0 +1,188 @@
+                                 sigma[2] * grad_u[7];
+
+            eshelby_stress[6] -= sigma[0] * grad_u[2] +
+                                 sigma[3] / std::sqrt(2) * grad_u[5] +
+                                 sigma[5] / std::sqrt(2) * grad_u[8];
+
+            eshelby_stress[7] -= sigma[3] / std::sqrt(2) * grad_u[2] +
+                                 sigma[1] * grad_u[5] +
+                                 sigma[4] / std::sqrt(2) * grad_u[8];
+
+            eshelby_stress[8] -= sigma[5] / std::sqrt(2) * grad_u[2] +
+                                 sigma[4] / std::sqrt(2) * grad_u[5] +
+                                 sigma[2] * grad_u[8];
+        }
+        else if (DisplacementDim == 2)
+        {
+            eshelby_stress[0] = eshelby_stress[DisplacementDim + 1] =
+                eshelby_stress[4] = psi;
+
+            eshelby_stress[0] -=
+                sigma[0] * grad_u[0] + sigma[3] / std::sqrt(2) * grad_u[2];
+
+            eshelby_stress[1] -=
+                sigma[3] / std::sqrt(2) * grad_u[0] + sigma[1] * grad_u[2];
+
+            eshelby_stress[2] -=
+                sigma[0] * grad_u[1] + sigma[3] / std::sqrt(2) * grad_u[3];
+
+            eshelby_stress[3] -=
+                sigma[3] / std::sqrt(2) * grad_u[1] + sigma[1] * grad_u[3];
+
+            // for axial-symmetric case the following is not zero in general
+            eshelby_stress[4] -= sigma[2] * grad_u[4];
+        }
+        else
+            OGS_FATAL(
+                "Material forces not implemented for displacement dimension "
+                "other than 2 and 3.");
+
+        auto const& w = _ip_data[ip].integration_weight;
+        local_b += G.transpose() * eshelby_stress * w;
+    }
+
+    return nodal_values;
+}
+
+template <typename LocalAssemblerInterface>
+void writeMaterialForces(
+    MeshLib::PropertyVector<double>& material_forces,
+    std::vector<std::unique_ptr<LocalAssemblerInterface>> const&
+        local_assemblers,
+    NumLib::LocalToGlobalIndexMap const& local_to_global_index_map,
+    GlobalVector const& x)
+{
+    DBUG("Compute material forces for small deformation process.");
+
+    // Material forces
+    std::fill(std::begin(material_forces), std::end(material_forces), 0);
+
+    GlobalExecutor::executeDereferenced(
+        [](const std::size_t mesh_item_id,
+           LocalAssemblerInterface& local_assembler,
+           const NumLib::LocalToGlobalIndexMap& dof_table,
+           GlobalVector const& x,
+           std::vector<double>& node_values) {
+            auto const indices = NumLib::getIndices(mesh_item_id, dof_table);
+            std::vector<double> local_data;
+            auto const local_x = x.get(indices);
+
+            local_assembler.getMaterialForces(local_x, local_data);
+
+            assert(local_data.size() == indices.size());
+            for (std::size_t i = 0; i < indices.size(); ++i)
+                node_values[indices[i]] += local_data[i];
```

Comment:
``indices`` contains the global indices. 

Just checked the dropped ProcessLib/SmallDeformationCommon/Common.h again, where nodal force vector was change to a global vector in the first commit of PR #1838. Hereby  ``node_values`` is a local one in partition.  Unfortunately, we do not store local indices of partition. Is it possible to change the type of ``node_values`` to GlobalVector?
 
PS: I remembered that we had discussed to change the property vector type to GlobalVector.

Reply:
The output is already tailed for the global vector, where the global values are copied to a local vector. See
https://github.com/ufz/ogs/blob/master/ProcessLib/ProcessOutput.cpp#L58